### PR TITLE
add padToMod in Chunk of bits

### PR DESCRIPTION
**Pull request by helanto**
Deals with requirement 7 of #5050

### CODE REVIEW

File: core-tests/shared/src/test/scala/zio/BitChunkSpec.scala [hunk 1]
```diff
@@ -11,6 +11,10 @@ object BitChunkSpec extends ZIOBaseSpec {
       bytes <- Gen.listOf(Gen.anyByte)
     } yield Chunk.fromIterable(bytes)
 
+  val genBitChunk: Gen[Random with Sized, Chunk[Boolean]] = for {
```
Comment (adamgfraser) [inline core-tests/shared/src/test/scala/zio/BitChunkSpec.scala:14]: Need to use `asBits` here to construct a `BitChunk`.

File: core/shared/src/main/scala/zio/Chunk.scala [hunk 2]
```diff
@@ -118,6 +118,23 @@ sealed abstract class Chunk[+A] extends ChunkLike[A] { self =>
...
+   * Pads to the specified number of bits.
+   *  - if n > size, chunk is 0-padded to length n.
+   *  - if n <= size, chunk is left intact.
+   *
+   * @example
+   * {{{
+   *  Chunk(true, false).padToMod(8) => Chunk(true, false, false, false, false, false, false, false)
+   * }}}
+   */
+  def padToMod(n: Int)(implicit ev: A <:< Boolean): Chunk[Boolean] =
```
Comment (adamgfraser) [inline core/shared/src/main/scala/zio/Chunk.scala:131]: This needs to have a lower level implementation on the `BitChunk` data types to have the performance we are looking for.

**Review by adamgfraser — CHANGES_REQUESTED** [review]
See comments below about using the `BitChunk` data type, which provides a more optimized representation of a chunk of bits than just a `Chunk[Boolean]`.

**Comment by jdegoes** [comment]
@helanto Will close for now. Please re-open if you get a chance to push this through!

**Pull request closed without merging**